community/contributing/pr_review_guidelines.rst

```diff
@@ -0,0 +1,396 @@
+     approving the PR, as appropriate).
+
+  3. An engine maintainer reviews the code and provides feedback, requests
+     changes, or approves the pull request, as appropriate.
+
+  4. Another maintainer reviews the code with a focus on code style/clarity and
+     approves it once satisfied.
+
+  5. A team leader or a member of the `production team
+     <https://godotengine.org/teams>`_ merges the pull request if satisfied that
+     it has been sufficiently reviewed.
+
+This document will explain steps 2, 3, 4, and 5 in more detail. For a more
+detailed explanation of the pull request workflow please see the :ref:`pull
+request workflow document <doc_pr_workflow>`. 
+
+.. note:: 
+  In practice these steps may blend together. Oftentimes maintainers will
+  provide comments on code style and code quality at the same time and will
+  approve a pull request for both.
+
+Typically the first interaction on a pull request will be an engine maintainer
+assigning tags to the pull request and flagging it for review by someone
+familiar with that area of code.
+
+Engine maintainers are folks who are "members" of the Godot project repository
+on Github and/or are listed on the `Teams page <https://godotengine.org/teams>`_
+on the Godot website. Maintainers are responsible for a given area of the
+engine. Typically this means they are the people who are given more trust to
+approve and recommend pull requests for merging.
+
+Even if you are not a maintainer, you can still help by reviewing code,
+providing feedback on PRs and testing PRs locally on your machine to confirm
+that they work as intended.
+
+Code review
+-----------
+
+The following is a list of things that contributors and engine maintainers can
+do to conduct a substantive code review of a pull request.
+
+.. note::
+  If you want to conduct a code review, but can't do everything on this list,
+  say that in your review comment. For example, it is still very helpful to
+  provide comments on code, even if you can't build the pull request locally to
+  test the new feature (or vice versa). Feel free to review the code, just
+  remember to make a note at the end of your review that you have reviewed the
+  code only and have not tested the changes locally.
+
+1. Confirm that the problem exists
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+PRs need to solve problems and problems need to be documented. Make sure that
+the pull request links and closes (or at least) addresses a bug or a proposal.
+If it doesn't, consider asking the contributor to update the opening message of
+the PR to explain the problem that the PR aims to solve in more detail.
+
+.. note::
+  It should be clear _why_ a pull request is needed before it is merged. This
+  assists reviewers in determining whether a PR does what it says it does and it
+  helps contributors in the future understand why the code is the way it is.
+
+2. Test the PR and look for regressions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Comment:
```suggestion
2. Test the PR and look for regressions
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Formatting issue.

Also, in your rework the part about testing the PR for its proposed functionality is kind of amiss. It was point 3 in my PR, but here point 3 is different, and I can't find any similar description.

I think we should still incorporate the message from my PR about testing the improved functionality here, in some form. Either as a part of this point 2, or in place of your point 3 (which I've already proposed to move elsewhere, so it all fits together nicely).

Reply:
I removed the previous section 3 intentionally as I didn't feel it added much value on top of the other sections (same thing with section 0).

Reply:
I get that testing the feature itself is kind of expected, but I think it would be good to cover even the obvious parts in this document. At least in passing, as we are making a checklist for maintainers to follow.

Reply:
Okay, added a note about ensuring the PR does what it does it does (fixes bug or implements feature)